edb/edgeql/compiler/normalization.py

```diff
@@ -42,6 +45,46 @@ def normalize(
     raise AssertionError(f'normalize: cannot handle {node!r}')
 
 
+def renormalize_compat(
+    norm_qltree: qlast.Base,
+    orig_text: str,
+    *,
+    schema: s_schema.Schema,
+    localnames: AbstractSet[str] = frozenset(),
+) -> qlast.Base:
+    """Renormalize an expression normalized with imprint_expr_context()."""
```

Comment:
Would help if you could clarify in this docstring what renormalization would actually do to it.

Reply:
Looks like this is needed for compatibility with pre a7 dumps?

Reply:
And it essentially re-creates the old WITH-style normalization?

Reply:
It takes the old `WITH-style` normalization and renormalizes it into the new explicit module name injection.

Reply:
So... can you add that to the docstring?